src/snowflake/snowpark/session.py

```diff
@@ -3218,10 +3271,33 @@ def flatten(
             - :meth:`DataFrame.flatten`, which creates a new :class:`DataFrame` by exploding a VARIANT column of an existing :class:`DataFrame`.
             - :meth:`Session.table_function`, which can be used for any Snowflake table functions, including ``flatten``.
         """
+        # AST.
+        stmt = self._ast_batch.assign()
+        expr = with_src_position(stmt.expr.sp_flatten, stmt)
+        build_const_from_python_val(input, expr.input)
+        if path is not None:
+            expr.path.value = path
+        expr.outer = outer
+        expr.recursive = recursive
+
+        mode = mode.upper()
+        if mode not in ("OBJECT", "ARRAY", "BOTH"):
+            raise ValueError("mode must be one of ('OBJECT', 'ARRAY', 'BOTH')")
```

Comment:
Why not raise on a last `else` clause at the end of the `if elif` chain? (Line 3293)